consumer/pdata/trace.go

```diff
@@ -130,29 +130,70 @@ const (
-// and is numerically equal to Standard GRPC codes https://github.com/grpc/grpc/blob/master/doc/statuscodes.md
+// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-status
 type StatusCode otlptrace.Status_StatusCode
 
 const (
-	StatusCodeOk                 = StatusCode(otlptrace.Status_STATUS_CODE_OK)
-	StatusCodeCancelled          = StatusCode(otlptrace.Status_STATUS_CODE_CANCELLED)
-	StatusCodeUnknownError       = StatusCode(otlptrace.Status_STATUS_CODE_UNKNOWN_ERROR)
-	StatusCodeInvalidArgument    = StatusCode(otlptrace.Status_STATUS_CODE_INVALID_ARGUMENT)
-	StatusCodeDeadlineExceeded   = StatusCode(otlptrace.Status_STATUS_CODE_DEADLINE_EXCEEDED)
-	StatusCodeNotFound           = StatusCode(otlptrace.Status_STATUS_CODE_NOT_FOUND)
-	StatusCodeAlreadyExists      = StatusCode(otlptrace.Status_STATUS_CODE_ALREADY_EXISTS)
-	StatusCodePermissionDenied   = StatusCode(otlptrace.Status_STATUS_CODE_PERMISSION_DENIED)
-	StatusCodeResourceExhausted  = StatusCode(otlptrace.Status_STATUS_CODE_RESOURCE_EXHAUSTED)
-	StatusCodeFailedPrecondition = StatusCode(otlptrace.Status_STATUS_CODE_FAILED_PRECONDITION)
-	StatusCodeAborted            = StatusCode(otlptrace.Status_STATUS_CODE_ABORTED)
-	StatusCodeOutOfRange         = StatusCode(otlptrace.Status_STATUS_CODE_OUT_OF_RANGE)
-	StatusCodeUnimplemented      = StatusCode(otlptrace.Status_STATUS_CODE_UNIMPLEMENTED)
-	StatusCodeInternalError      = StatusCode(otlptrace.Status_STATUS_CODE_INTERNAL_ERROR)
-	StatusCodeUnavailable        = StatusCode(otlptrace.Status_STATUS_CODE_UNAVAILABLE)
-	StatusCodeDataLoss           = StatusCode(otlptrace.Status_STATUS_CODE_DATA_LOSS)
-	StatusCodeUnauthenticated    = StatusCode(otlptrace.Status_STATUS_CODE_UNAUTHENTICATED)
+	StatusCodeUnset = StatusCode(otlptrace.Status_STATUS_CODE_UNSET)
+	StatusCodeOk    = StatusCode(otlptrace.Status_STATUS_CODE_OK)
+	StatusCodeError = StatusCode(otlptrace.Status_STATUS_CODE_ERROR)
 )
 
 func (sc StatusCode) String() string { return otlptrace.Status_StatusCode(sc).String() }
+
+// SetCode replaces the code associated with this SpanStatus.
+//
+// Important: This causes a runtime error if IsNil() returns "true".
+func (ms SpanStatus) SetCode(v StatusCode) {
+	(*ms.orig).Code = otlptrace.Status_StatusCode(v)
+
+	// According to OTLP spec we also need to set the deprecated_code field.
+	// See https://github.com/open-telemetry/opentelemetry-proto/blob/59c488bfb8fb6d0458ad6425758b70259ff4a2bd/opentelemetry/proto/trace/v1/trace.proto#L231
+	//
+	//   if code==STATUS_CODE_UNSET then `deprecated_code` MUST be
+	//   set to DEPRECATED_STATUS_CODE_OK.
+	//
+	//   if code==STATUS_CODE_OK then `deprecated_code` MUST be
+	//   set to DEPRECATED_STATUS_CODE_OK.
+	//
+	//   if code==STATUS_CODE_ERROR then `deprecated_code` MUST be
+	//   set to DEPRECATED_STATUS_CODE_UNKNOWN_ERROR.
+	switch v {
+	case StatusCodeUnset:
+		fallthrough
+	case StatusCodeOk:
+		ms.SetDeprecatedCode(DeprecatedStatusCodeOk)
```

Comment:
Just merge these 2 cases instead of using fallthrough?

`case StatusCodeUnset, StatusCodeOk:`

Reply:
Fixed.